template/template.go

```diff
@@ -186,7 +186,7 @@ func NewTemplate(i *NewTemplateInput) (*Template, error) {
 	}
 
 	// Compute the MD5, encode as hex
-	hash := md5.Sum([]byte(t.contents))
+	hash := md5.Sum([]byte(t.source + t.contents + t.destination))
```

Comment:
It probably would be a good idea to add null bytes between each of these just to make it robust against very silly inputs.

Should we swallow the pill and swap this out for a hash function that doesn't trip zealous security people while we're here? (ex. sha256, which also has hardware support on some architectures.) I'm fine with the answer being "not in this PR" 😁